### ISSUE TITLE
no icons in tab bar

### ISSUE THREAD

**Issue by ghost** [alcinnz/Odysseus#69]
i think this is more inline with other elementary apps like files and scratch.

in its place could be a notification icon of sorts that lets you know if the web pages content has changed since last you looked at the tab

**Comment by alcinnz** [comment]
How about making those icons more grayscale, to fit in better with text?

I find the icons useful for telling tabs apart at a quick glance, especially when those tabs get pinned.

**Comment by ujon** [comment]
i personally find them obtrusive... would there be an easy way to disable them only in the tab bar and not globally i.e. in the address bar?

**Comment by alcinnz** [comment]
Sure that'd be no problem. The bigger problem is the icons are quite vital for "pinned" tabs, and disabling them for other tabs except the pinned ones is the tricky bit. But I can see even for those pinned tabs, the icons can look a bit obtrusive.

So I'd want to start by making those icons look less obtrusive for the pinned tabs, before maybe hiding them entirely if they're still too obtrusive.

**Comment by alcinnz** [comment]
The good news for that approach is that most favicons are monochromatic. That means it should be relatively easy to make them blend in with text whilst maintaining recognizability.

**Comment by ujon** [comment]
i dont (currently) have any tabs pinned but i see where you are coming from

**Comment by alcinnz** [comment]
Just looked into the APIs available to me, and this transform looks surprisingly easy!

GDK pixbuf makes it easy to perform a grayscale transform https://valadoc.org/gdk-pixbuf-2.0/Gdk.Pixbuf.saturate_and_pixelate.html, and Cairo makes it easy to lighten the image https://valadoc.org/cairo/Cairo.Operator.LIGHTEN.html (which'll be needed to blend in with elementary's off-black text colour https://github.com/elementary/stylesheet/blob/master/gtk-3.0/gtk.css#L30).

**Comment by alcinnz** [comment]
I've now got a nice script I can integrate which'll recolour favicons to the elementary's text colour. Integrating it today.

**Comment by alcinnz** [comment]
Sorry it's taken me so long, but I've just committed a fix for this issue.

I find interesting to note that to obtain unobtrusiveness it wasn't enough to match elementary's text colour, simply because favicons generally fill in many more pixels than a (opensans) character. So I ended up going for the lightest black in elementary's colour pallette.

I hope you like this fix, it should hopefully be live tomorrow.
